pkg/inject/inject.go

```diff
@@ -599,6 +600,15 @@ func (conf *ResourceConfig) injectPodSpec(patch *Patch) {
 		},
 	}...)
 
+	if !conf.tapDisabled() {
```

Comment:
i think this should move up to near the other `conf.tapDisabled()` check, for readability, and also because there's an early return on line 557 in the case where identity is disabled.